app/scripts/directives/mobileClientBuildsList.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileClientBuildsList', {
+    controller: [
+      '$timeout',
+      '$anchorScroll',
+      '$location',
+      'APIService',
+      'DataService',
+      MobileClientBuildsList
+    ],
+    bindings: {
+      mobileClient: '<'
+    },
+    templateUrl: 'views/mobile-client-builds-list.html'
+  });
+
+  function MobileClientBuildsList(
+    $timeout,
+    $anchorScroll,
+    $location,
+    APIService,
+    DataService
+  ) {
+    var ctrl = this;
+    var buildConfigsVersion = APIService.getPreferredVersion('buildconfigs');
+    var watches = [];
+    var anchor;
+
+    ctrl.$onChanges = function(changes) {
+      var mobileClientChanged = _.get(changes, "mobileClient.currentValue", false);
+      if (mobileClientChanged && !ctrl.context) {
+        ctrl.context = {namespace: _.get(ctrl, 'mobileClient.metadata.namespace')};
+      }
+
+      if (mobileClientChanged && !ctrl.buildsWatchSet) {
+        ctrl.buildsWatchSet = true;
+        watches.push(DataService.watch(buildConfigsVersion, ctrl.context, function(buildConfigs) {    
+          ctrl.mobileBuildConfigs = _.filter(buildConfigs.by('metadata.name'), function(buildConfig) {
+            return _.get(buildConfig, 'metadata.labels.mobile-client-build') === 'true' && _.get(buildConfig, 'metadata.labels.mobile-client-id') === _.get(ctrl, 'mobileClient.metadata.name');
+          });
+          ctrl.loaded = true;
+          ctrl.goToBuildConfig();
+        }));
+      }
+    };
+
+    ctrl.goToBuildConfig = function() {
+      if ($location.hash() && anchor !== $location.hash()) {
+        anchor = $location.hash();
+
+        $timeout(function() {
+          $anchorScroll.yOffset = 100;
```

Comment:
What is the significance of `100`?  I assume `px` value, and that this could be thrown off by CSS changes.  Could this be a named variable giving us a better clue what its relationship is to the UI?

Reply:
This was the height of the header on the page. We need this offset to scroll our components correctly. I have renamed the variable to suit.

Reply:
Do we do this kind of scrolling elsewhere?

Perhaps also comment on the specific CSS class name that it matches.  This kind of thing can very easily break when there are CSS changes. 

Possible to use `element` within the `directive` and get the actual `y` coords of the component you want to navigate to && do a direct scroll?   Not sure how viable this is within your context.